src/Artifacts.jl

```diff
@@ -283,53 +295,91 @@ function download_artifact(
+            # and Pkg.jl#2317, we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the
+            # error and move the artifact to the expected location and return true
+            ignore_hash_env = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", false)
+            if ignore_hash_env || ignore_hash_symlink
+                if ignore_hash_symlink
+                    msg *= """
+
+                    Note: Julia cannot create symlinks, which may be the reason for the hash mismatch.
+                    One solution is to activate Developer Mode in Windows, for instructions on how to do that, see:
+                        https://learn.microsoft.com/en-us/gaming/game-bar/guide/developer-mode
+                    An alternative path is to ignore the artifact hash mismatches, 
+                    however this is generally not recommended.
+
+                    Symlinks that could not be extracted:
+                    $(sprint(io->show(io, MIME"text/plain"(), changed_symlinks_output)))
+                    """
+                end
+                if ignore_hash_env
+                    msg *= "\$JULIA_PKG_IGNORE_HASHES is set to 1:\n"
+                end
+                msg *= "\nIgnoring error and moving artifact to the expected location"
+                @error(msg)
+                # Move it to the location we expected
+                _artifact_rename(new_path, temp_dir)
+            else
+                error(msg)
+            end
         end
+        # Move it to the location we expected
+        _artifact_rename(new_path, temp_dir)
     catch err
         @debug "download_artifact error" tree_hash tarball_url tarball_hash err
         if isa(err, InterruptException)
             rethrow(err)
         end
         # If something went wrong during download, return the error
         return err
+    finally
+        # Always attempt to cleanup
+        rm(temp_dir; recursive=true, force=true)
     end
-
-    # Did we get what we expected?  If not, freak out.
-    if calc_hash.bytes != tree_hash.bytes
-        msg  = "Tree Hash Mismatch!\n"
-        msg *= "  Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))\n"
-        msg *= "  Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))"
-        # Since tree hash calculation is still broken on some systems, e.g. Pkg.jl#1860,
-        # and Pkg.jl#2317, we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the
-        # error and move the artifact to the expected location and return true
-        ignore_hash = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", false)
-        if ignore_hash
-            msg *= "\n\$JULIA_PKG_IGNORE_HASHES is set to 1: ignoring error and moving artifact to the expected location"
-            @error(msg)
-            # Move it to the location we expected
-            src = artifact_path(calc_hash; honor_overrides=false)
-            dst = artifact_path(tree_hash; honor_overrides=false)
-            mv(src, dst; force=true)
```

Comment:
If `calc_hash` was an existing valid artifact, that artifact gets deleted by calling `mv` here. That's why I reordered things and removed the call to `create_artifact`.